nexus/reconfigurator/planning/src/blueprint_builder/builder.rs

```diff
@@ -1999,14 +1997,18 @@ impl<'a> BlueprintBuilder<'a> {
     /// Try to find an artifact in either the current or previous release repo
     /// that contains an image for a zone of the given kind; see RFD 565 §9.
     /// Defaults to the install dataset.
+    ///
+    /// Takes an optional "zone_id" as input, to avoid downgrading zones which
+    /// have already committed to using the new repo.
     pub(crate) fn zone_image_source(
         &self,
+        zone_id: Option<OmicronZoneUuid>,
```

Comment:
Can we make this parameter mandatory and always do the check in `zone_should_use_new_repo`? It seems confusing to have to decide at each call site whether to pass the `zone_id` in or not.

Reply:
Sure thing; I didn't know if the `zone_id` would be accessible everywhere, but it seems like it is.

Agreed this is better as a non-Optional argument.